modules/resolve.js

```diff
@@ -0,0 +1,83 @@
+export const resolveLocation = (location, base) => {
+  if (typeof location === 'string') {
+    return resolve(location, base)
+  } else {
+    location.pathname = resolve(location.pathname, base)
```

Comment:
We're already using https://github.com/mjackson/resolve-pathname in the history lib. I think it makes sense to re-use it here?